### ISSUE TITLE
Added IRC Chat in the client

### ISSUE THREAD

**Issue by FuzzyBearBTC** [Peerunity/Peerunity#161]
https://github.com/FuzzyBearBTC/Peerunity been adding IRC chat and a block explorer to the client, chat is working BE still has bugs

**Comment by brossi** [comment]
@FuzzyBearBTC - I appreciate the initiative you've taken, thank you. I can see how a block explorer would be a very useful tool to include as long as it doesn't impact performance of the client in a negative way.

I'll be honest, though... the thought of adding IRC functionality into a mission critical financial application makes me very uneasy. Has this feature been one that has been widely requested and is there a production-enabled example of it in another project's wallet client that could be used as a reference point?

**Comment by Thireus** [comment]
I think @FuzzyBearBTC's feature is interesting. However I'm also with @brossi on that. A wallet shouldn't do more than what it has been designed for for obvious security reasons. However I think Peerunity already connects to IRC to disclose nodes (which btw isn't working anymore), isn't it? :)

**Comment by FuzzyBearBTC** [comment]
@brossi The idea came from looking at other coin clients and what features they had.  Digitalcoin was the one that I found with IRC, blockexplorer and market data tabs, and what I have used as template to work with peerunity.  
blockexplorer adding commit https://github.com/DGCFoundation/digitalcoin/commit/1ea07b1d6122b37709936e65f58d4d26ead9da29
adding IRC chat and market data commit https://github.com/DGCFoundation/digitalcoin/commit/e9d5a80a2d5459d7a24c761974708c35f4a9bf6d

I agree completely with your concerns and I would not be using this on any minting device or wallet I hold a lot of coins on but is useful for me and would like others to look at the code and clarify any vulnerabilities etc 

Other coins have also implemented the feature and i am yet to see any issues raised from their communities:
DayTraderCoin - https://bitcointalk.org/index.php?topic=827984.msg9789836;topicseen#msg9789836

**Comment by FuzzyBearBTC** [comment]
@brossi @Thireus I would really like to understand if we could make this a custom build option on the .pro file? so we pass in an argument on the qmake and this sets the IRC build on otherwise it is disabled?

**Comment by emeth-** [comment]
@FuzzyBearBTC This would be really sweet to embed in PeerApps - both the block explorer and the IRC client. It's kind of what PeerApps is meant to be - a collection of applications for the community, centered around Peercoin, that runs separate from the wallet (but adds features on top of it). They don't all have to be decentralized. I intend on including (for example) findstake.js.

**Comment by FuzzyBearBTC** [comment]
@emeth- yes that sounds like a suitable suggestion for PeerApps and I think would most likely be more suitable to be located in PeerApps and not the main client.  Happy to help port this across to PeerApps when ready to, just give me a PM or mention on here.
